azure-pipelines.yml

```diff
@@ -121,13 +121,13 @@ stages:
       displayName: 'Setup Python $(PYTHON_VERSION)'
     - script: |
         python -m pip install --progress-bar off --upgrade pip setuptools wheel
-        python -m pip install --progress-bar off .[test]
-      displayName: 'Install dependencies and MNE (stable)'
+        python -m pip install --progress-bar off --upgrade .[test]
+      displayName: 'Install package'
     - script: |
-        python -m pip uninstall -y mne
-        python -m pip install --progress-bar off --no-deps git+https://github.com/mne-tools/mne-python
-      condition: eq(variables.MNE_DEV, 'true')
-      displayName: 'Install MNE (main)'
+        python -m pip install --progress-bar off --force-reinstall git+https://github.com/mne-tools/mne-python
+        python -m pip install --progress-bar off --upgrade --pre --only-binary :all: -i https://pypi.anaconda.org/scientific-python-nightly-wheels/simple --timeout=180 numpy scipy scikit-learn matplotlib
```

Comment:
That's the one test I changed. Instead of installing only MNE (main), I also install the pre-release version of numpy, scipy, scikit-learn and matplotlib (they need each other to work here, we can't mix a pre-release numpy with the current stable matplotlib or scikit-learn).